python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/resource.py

```diff
@@ -73,6 +73,9 @@ def extract_resource_metadata(
         # shared metadata that is displayed for all assets
         base_metadata = {k: v for k, v in load_info_dict.items() if k in dlt_base_metadata_types}
 
+        if resource is None:
```

Comment:
Why would resource be optional now?

Reply:
It should never be the case, however, I was getting a type error of it being None when retrieved from the metadata dictionary. There may be a cleaner way of doing it, and I'm open to suggestions.

```
metadata = self.extract_resource_metadata(dlt_source_resource, load_info)

■ Argument of type "Any | None" cannot be assigned to parameter "resource" of type "DltResource" in function "extract_resource_metadata"    Type "Any | None"
```

Reply:
You could do 
```
asset dlt_source_resource is not None
```

or fixing the metadata dictionary so that it returns a more explicit type if possible

Reply:
Let's use `dagster._check` to narrow the type

Reply:
Added here - thanks. https://github.com/dagster-io/dagster/pull/20785/commits/e9e679f24cc76f204016d5236a4bae57719e4b7d

Reply:
Ok, ended up moving the translator to the metadata, which will give us more flexibility in the future. Also removed my None checks with `check.inst` to clean things up! Thank you both.

https://github.com/dagster-io/dagster/pull/20785/commits/db7683734d7e6923bcae1c3fda5625b4844118fd